### PR TITLE
Fix DuckDB bundling and add support for DuckDB install on MacOS

### PR DESCRIPTION
**Pull request by majetideepak**
DuckDB fails to build after the recent support for INSTALL_PREFIX

Resolves https://github.com/facebookincubator/velox/issues/11058

### CODE REVIEW

**Comment by majetideepak** [comment]
@yingsu00 Can you verify if this fixes your issue?

File: CMakeLists.txt [hunk 1]
```diff
@@ -192,6 +189,18 @@ if(${VELOX_BUILD_PYTHON_PACKAGE})
   set(VELOX_ENABLE_SPARK_FUNCTIONS ON)
 endif()
 
+if(${VELOX_ENABLE_DUCKDB})
```
Comment (yingsu00) [inline CMakeLists.txt:192]: Does this need to happen after line 48? Is it possible to unify the two `if(DEFINED ENV{INSTALL_PREFIX})` blocks? If not, will you please add a comment? Thanks.

Comment (assignUser) [reply CMakeLists.txt:192]: `include_directories` modifies a global property which is applied to targets at the time of creation. So moving it to after duckdb is resolved keeps the changes away from it.

**Review by assignUser — APPROVED** [review]
Thanks for the fix.

In the future maybe a target based approach applied to only velox targets would be better, we can try that in tandem with the planned changes to flags and warnings.

File: CMakeLists.txt (hunk 1 above)
Comment (zuyu) [reply CMakeLists.txt:192]: @yingsu00 Yes, your proposal that moving `CMAKE_PREFIX_PATH` setup along w/ `include_directoryies` works.

```cmake
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -45,14 +45,6 @@ if(DEFINED ENV{CONDA_PREFIX})
   endif()
 endif()

-if(DEFINED ENV{INSTALL_PREFIX})
-  message(STATUS "Dependency install directory set to: $ENV{INSTALL_PREFIX}")
-  list(APPEND CMAKE_PREFIX_PATH "$ENV{INSTALL_PREFIX}")
-  # Allow installed package headers to be picked up before brew/system package
-  # headers
-  include_directories(BEFORE "$ENV{INSTALL_PREFIX}/include")
-endif()
-
 list(PREPEND CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/CMake"
      "${PROJECT_SOURCE_DIR}/CMake/third-party")

@@ -192,6 +184,21 @@ if(${VELOX_BUILD_PYTHON_PACKAGE})
   set(VELOX_ENABLE_SPARK_FUNCTIONS ON)
 endif()

+if(${VELOX_ENABLE_DUCKDB})
+  set_source(DuckDB)
+  resolve_dependency(DuckDB)
+endif()
+
+if(DEFINED ENV{INSTALL_PREFIX})
+  message(STATUS "Dependency install directory set to: $ENV{INSTALL_PREFIX}")
+  list(APPEND CMAKE_PREFIX_PATH "$ENV{INSTALL_PREFIX}")
+
+  # Allow installed package headers to be picked up before brew/system package
+  # headers
+  include_directories(BEFORE "$ENV{INSTALL_PREFIX}/include")
+endif()
```

And the output log:

```cmake
-- Setting DuckDB source to AUTO
-- Building DuckDB from source
-- Using BUNDLED DuckDB
-- Dependency install directory set to: /velox_dependency_install
-- Found OpenSSL: /opt/homebrew/Cellar/openssl@3/3.3.2/lib/libcrypto.dylib (found version "3.3.2")
-- Using ccache: /opt/homebrew/bin/ccache
```

Comment (assignUser) [reply CMakeLists.txt:192]: This will prevent an installed duckdb from being found and always build it from source, so not what we want :)

**Comment by assignUser** [comment]
Once @yingsu00 confirms this fixes the issue this is `ready-to-merge`

File: CMakeLists.txt (hunk 1 above)
Comment (zuyu) [reply CMakeLists.txt:192]: Make sense. I never thought about installing duckdb in `INSTALL_PREFIX`, but we do not install duckdb in macOS setup script, right?

Comment (assignUser) [reply CMakeLists.txt:192]: After this PR duckdb will be installed with the setup-macos :) https://github.com/facebookincubator/velox/pull/11069/files#diff-49efb0acd7ee0e6a2d6f57f2e48b90bd170a41a7c46d2ef757a6c078b543d28cR151d

**Comment by yingsu00** [comment]
> Once @yingsu00 confirms this fixes the issue this is `ready-to-merge`

It works for me locally with latest folly.

**Review by czentgr — APPROVED** [review]

**Pull request closed without merging**